Munchies.cs

```diff
@@ -29,112 +29,85 @@ public override void Unload() {
 			ToggleReportHotKey = null;
 		}
 
-		// How to add external content to the list: do the following in YOUR mod in PostSetupContent
-		//Mod munchiesMod = ModLoader.GetMod("Munchies");
-		//if (munchiesMod != null)
-		//{
-		//	munchiesMod.Call(<arguments array>);
-		//}
-
-		/*
-		 * string[] args = {
-				"AddMod",
-				"Calamity",
-				"CalamityMod/icon_small"
-			};
-			munchiesMod.Call(args);
-		 * object[] bloodOrangeArgs = {
-				"AddConsumable",
-				"Calamity",
-				"Blood Orange",
-				"CalamityMod/Items/PermanentBoosters/BloodOrange",
-				"Increases max health by 25 if user has at least 500 HP",
-				"player_normal",
-				new Func<bool>(GetBOrange),
-				"38",
-				"40"
-			};
-			munchiesMod.Call(bloodOrangeArgs);
-		 * 
-
-		/*
-		 * 0: Function: AddMod => string
-		 * 0: Mod tab name => string
-		 * 1: Mod tab texture => string
-		 * 
-		 * 0: Function: AddConsumable => string
-		 * 1: Consumable mod => string
-		 * 2: Consumable name => string
-		 * 3: Consumable texture => string
-		 * 4: Consumable hover text => string
-		 * 5: Consumable type => string OR a Color object
-		 * 6: Consumable hasBeenConsumed => Func<bool>
-		 * 7: Asset dimensions X => float
-		 * 8: Asset dimensions Y => float
-		 */
 		public override object Call(params object[] args) {
 			try {
 				if (args.Length >= 1) {
 					return args[0] switch {
-						"AddMod" => HandleAddMod(args),
 						"AddConsumable" => HandleAddConsumable(args),
+						"AddVanillaConsumable" => HandleAddVanillaConsumable(args),
+						// can add more types of calls here in the future
 						_ => false
 					};
 				} else {
-				return false;
-			}
+					return false;
+				}
 			} catch (Exception e) {
 				Logger.Error($"Call Error: {e.StackTrace} {e.Message}");
 			}
 			return "Failure";
 		}
 
-		private bool HandleAddMod(params object[] args) {
-			if (args.Length == 3) {
-				ConsumableMod mod = new(modTabName: args[1] as string, modTabTexturePath: args[2] as string);
+		private bool HandleAddConsumable(params object[] args) {
+			Mod mod = null;
+			try {
+				Consumable consumable;
+				mod = args[1] as Mod;
+				ConsumableMod externalMod = new(mod);
+
+				object apiString = args[2];
+				Version apiVersion = apiString is string ? new Version(apiString as string) : this.Version; // current as of this update is 1.3
+				ModItem item = args[3] as ModItem;
+				Func<int> currentCount = args[5] as Func<int>;
+				Func<int> totalCount = args[6] as Func<int>;
 
-				return Report.AddModtoList(mod);
-			} else {
+				consumable = new(
+					modItem: item,
+					CategoryOrCustomColor: args[4],
+					currentCount: currentCount,
+					totalCount: totalCount
+				);
+
+				return Report.AddConsumableToList(mod: externalMod, consumable: consumable);
+
+				// Add optional fields for max # of items, option of func<int> for how many have been consumed
+			} catch (Exception e) {
+				(mod ?? this).Logger.Error($"Error adding consumable: {e.Message}, {e.StackTrace}");
```

Comment:
Two things:
1. The logger should still originate from your own mod always (`this`), since you invoke the logging, and the mod's Name should be used in the log message to indicate/point to it
2. Logging the exception's data is redundant as TML hooks into the exception logging which bypasses catching, so it always prints these out